app/components/chat/BaseChat.tsx

```diff
@@ -175,6 +175,15 @@ export const BaseChat = React.forwardRef<HTMLDivElement, BaseChatProps>(
         try {
           parsedApiKeys = getApiKeysFromCookies();
           setApiKeys(parsedApiKeys);
+
+          // If we have a provider selected, set its API key
+          if (provider) {
+            const savedKey = parsedApiKeys[provider.name] || '';
+
+            if (savedKey) {
+              onApiKeysChange(provider.name, savedKey);
+            }
+          }
         } catch (error) {
```

Comment:
is this section really required?, we are already saving the keys in the above line.

and anytime any apikey changes `onApiKeysChange ` is called anyway from below 
```typescript
<APIKeyManager
  provider={provider}
  apiKey={apiKeys[provider.name] || ''}
  setApiKey={(key) => {
    onApiKeysChange(provider.name, key);
  }}
/>
```

the key is stores as an object so does not matter if the provider changed, during api call it pulls the correct apiKey from the object for that provider